packages/idyll-cli/bin/idyll.js

```diff
@@ -1,90 +1,19 @@
-    d: 'datasets',
-    q: 'default-components',
-    f: 'input-file',
-    s: 'input-string',
-    l: 'layout',
-    n: 'no-minify',
-    o: 'output',
-    p: 'port',
-    r: 'no-ssr',
-    t: 'template',
-    e: 'theme',
-    w: 'watch'
-  })
-  .describe('components', 'Directory where components are located')
-  .default('components', 'components')
-  .array('components')
-  .describe('static', 'Directory where static assets are located')
-  .default('static', 'static')
-  .describe('css', 'Custom CSS file to include in output')
-  .describe('datasets', 'Directory where data files are located')
-  .default('datasets', 'data')
-  .describe('default-components', 'Directory where default set of components are located')
-  .array('default-components')
-  .describe('input-file', 'File containing Idyll source')
-  .describe('input-string', 'Idyll source as a string')
-  .describe('layout', 'Name of (or path to) the layout to use')
-  .default('layout', 'blog')
-  .boolean('no-minify')
-  .describe('no-minify', 'Skip JS minification')
-  .describe('output', 'Directory where built files should be written')
-  .default('output', 'build')
-  .boolean('no-ssr')
-  .describe('no-ssr', 'Do not pre-render HTML as part of the build')
-  .describe('template', 'Path to HTML template')
-  .array('transform')
-  .describe('transform', 'Custom browserify transforms to apply.')
-  .default('transform', [])
-  .describe('port', 'Custom port to bind the local server to.')
-  .describe('theme', 'Name of (or path to) the theme to use')
-  .default('theme', 'idyll')
-  .boolean('watch')
-  .describe('watch', 'Monitor input files and rebuild on changes')
-  .default('watch', false)
-  .describe('open', 'Automatically open the page in the default browser during watch mode.')
-  .default('open', true)
-  .alias('h', 'help')
-  .argv;
-
-// inputFile can be passed as non-hypenated argument
-if (argv._[0]) argv.f = argv.inputFile = argv._[0];
-
-// API checks the inverse
-argv.minify = !argv['no-minify'];
-argv.ssr = !argv['no-ssr'];
-
-// delete stuff we don't need
-delete argv._;
-delete argv['$0'];
-delete argv.n;
-delete argv.noMinify;
-delete argv['no-minify'];
-delete argv.k;
-delete argv.r;
-delete argv.noSsr;
-delete argv['no-ssr'];
-
-// delete undefined keys so Object.assign won't use them
-Object.keys(argv).forEach((key) => {
-  if (argv[key] === undefined) delete argv[key];
-})
-
-debug('Using CLI arguments:', argv)
-
-idyll(argv).build().on('complete', () => {
-  if (!argv.watch) {
-    process.exit(0);
-  }
+// Lightly modified from the Ember CLI
+// (https://github.com/ember-cli/ember-cli/blob/master/bin/ember)
+
+const p = require('path');
+const { spawnSync } = require('child_process');
+const { getLocalIdyll } = require('./util');
+
+const idyll = getLocalIdyll()
+
+var cmd
+if (!idyll) {
+  cmd = p.join(__dirname, './cli.js');
+} else {
+  cmd = p.join(idyll, '..', '..', 'bin', 'cli.js');
+}
+spawnSync(cmd, process.argv.slice(2), {
+  stdio: 'inherit'
```

Comment:
This seems like a great way of handling this issue. I'll try it out on a couple projects locally once tests are passing.